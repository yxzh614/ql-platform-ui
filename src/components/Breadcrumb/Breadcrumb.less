.ql-breadcrumb {
    .breadcrumb-item {
        display: inline-block;
        .breadcrumb-label {
            user-select: none;
            &:hover {
                color: rgb(96, 212, 212);
                &:active {
                    color: rgb(144, 228, 228);
                }
            }
        }
        .breadcrumb-seperator {
            user-select: none;
            margin: 0 4px;
        }
        .breadcrumb-seperator-hide {
            display: none;
        }
    }
}