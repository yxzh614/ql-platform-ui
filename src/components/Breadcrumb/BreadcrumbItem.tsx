import React from 'react'
import classnames from 'classnames'
import { FunctionComponent } from 'react'

export interface BreadcrumbItemProps {
    onClick?: (e: React.MouseEvent, value: string) => void
    classNames?: string
}
const BreadcrumbItem:FunctionComponent<BreadcrumbItemProps> = ({ children, onClick, classNames, ...others }) => {
    const handleClick:React.MouseEventHandler = (e) => {
        onClick && onClick(e, '123')
    }
    return (
        <span className={classnames('breadcrumb-label', classNames)} onClick={handleClick} {...others}>{children}</span>
    )
}
export default BreadcrumbItem