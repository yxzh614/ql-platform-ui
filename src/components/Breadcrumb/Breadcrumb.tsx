import React from 'react'
import './Breadcrumb.less'
import classnames from 'classnames'
import { Icon } from '../..'
import { number } from 'prop-types';
export interface BreadcrumbItem {
    label: string
    key: string
}

export interface BreadcrumbProps {
    datasource: Array<BreadcrumbItem>
    onClick: (e: React.MouseEvent, key: string) => void
    seperator?: string
}

export default class extends React.Component<BreadcrumbProps> {

    handleClick = (e: React.MouseEvent, key: string) => {
        this.props.onClick && this.props.onClick(e, key)
    }

    renderBreadcrumbItems () {

        const { datasource, seperator } = this.props

        if (datasource && datasource.length > 0) {
            return datasource.map((item, key) => {
                return (
                <div className={'breadcrumb-item'} key={key}>
                    <span className={classnames(
                        'breadcrumb-seperator',
                        key === 0 ? 'breadcrumb-seperator-hide' : ''
                        )}>{seperator ? seperator : (<Icon type={'right'}/>)}</span>
                    <span className="breadcrumb-label" onClick={(e) => {this.handleClick(e, item.key)}}>{item.label}</span>
                </div>
                )
            })
        }
        return 
    }

    render () {
        return (
            <div className="ql-breadcrumb">
                {this.renderBreadcrumbItems()}
            </div>
        )
    }
}