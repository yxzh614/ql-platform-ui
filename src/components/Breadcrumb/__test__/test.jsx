
import React from 'react'
import Breadcrumb from '../index'
import renderer from 'react-test-renderer'
const datasource = [{
    label: '主页',
    key: 'r1'
}, {
    label: '个人信息',
    key: 'r2'
}, {
    label: '修改密码',
    key: 'r3'
}]

test('Breadcrumb', () => {
    const component = renderer.create(
        <Breadcrumb datasource={datasource} onClick={(e, key) => {console.log(key)}}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });