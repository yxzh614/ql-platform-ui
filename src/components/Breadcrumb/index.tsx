import Breadcrumb from './Breadcrumb'

export {
    Breadcrumb
}
export default Breadcrumb