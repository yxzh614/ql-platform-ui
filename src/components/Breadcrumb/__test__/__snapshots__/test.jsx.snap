// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb 1`] = `
<div
  className="ql-breadcrumb"
>
  <div
    className="breadcrumb-item"
  >
    <span
      className="breadcrumb-seperator breadcrumb-seperator-hide"
    >
      /
    </span>
    <span
      className="breadcrumb-label"
      onClick={[Function]}
    >
      主页
    </span>
  </div>
  <div
    className="breadcrumb-item"
  >
    <span
      className="breadcrumb-seperator"
    >
      /
    </span>
    <span
      className="breadcrumb-label"
      onClick={[Function]}
    >
      个人信息
    </span>
  </div>
  <div
    className="breadcrumb-item"
  >
    <span
      className="breadcrumb-seperator"
    >
      /
    </span>
    <span
      className="breadcrumb-label"
      onClick={[Function]}
    >
      修改密码
    </span>
  </div>
</div>
`;
